gcp/appengine/utils.py

```diff
@@ -23,3 +27,43 @@ def is_prod():
   # and it's only used for Redis cache (which has its own env vars) and logging.
   # Consider removing this altogether.
   return 'GAE_ENV' in os.environ or 'K_SERVICE' in os.environ
+
+
+def relative_time(value: datetime | str) -> str:
+  """
+    Convert a datetime or ISO 8601 string to a human-readable relative time.
+
+    Args:
+        value: The input datetime as an ISO 8601 string or a datetime object.
+
+    Returns:
+        str: A human-readable string representing the relative time as:
+             - < 1 minute: Just now
+             - < 1 hour: x minutes ago
+             - < 1 day: x days ago
+             - < 2 day: yesterday
+             - < 7 days: x days ago
+             - >= 7 days: DD MMM
+             - Year is not current year: DD MMM YYYY
+    """
+  # Parse the ISO 8601 string to a datetime object
+  if isinstance(value, str):
+    value = isoparse(value)
+
+  now = datetime.now(tz=tz.tzutc())
+  diff = relativedelta(now, value)
+  diff_seconds = (now - value).total_seconds()
```

Comment:
```suggestion
  diff = now - value
  diff_seconds = diff.total_seconds()
  diff_minutes = diff_seconds // 60
  diff_hours = diff_seconds // (60 * 60)
```

I think we can change this part to avoid needing to import a new library.